src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1292,7 +1293,62 @@ public void ShowUpdatePackages(ShowUpdatePackageOptions updatePackageOptions)
                     SelectMatchingUpdatePackages(updatePackageOptions);
                 };
                 _packageList.LoadItemsCompleted += handler;
-            }
+            }  
+        }
+
+        public void AutomaticallySelectFirstPackage(string packageVersion)
+        {
+            _topPanel.SelectFilter(ItemFilter.All);
+            SelectedSource = PackageSources.FirstOrDefault(psm => psm.IsAggregateSource);
+            EventHandler handler = null;
+            handler = (s, e) =>
+            {
+                _packageList.LoadItemsCompleted -= handler;
+
+                _packageList.UpdateSelectedItem((PackageItemViewModel)_packageList.Items.FirstOrDefault());
```

Comment:
I'm going to stop my review here for now, and it's a bit late so I may not be thinking at my best....
but I think a lot of this code is duplicating existing code. See these methods:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs#L192
https://github.com/NuGet/NuGet.Client/blob/c1a0597e5498bcf3bf0938b3dd0170edbf0b2d20/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L1030